bundle/src/main/java/com/cognifide/cq/cqsm/core/progress/ProgressImpl.java

```diff
@@ -68,8 +62,7 @@ public void addEntry(Message message, Status status) {
 
 	@Override
 	public boolean isSuccess() {
-		calculateSuccess();
-		return success;
+		return ProgressHelper.calculateSuccess(entries);

```

Comment:
why did you remove the boolean field ? Now, with every call to isSuccess, the list will be iterated. Is there async processing performed, so the status of entries in the list can change ?

Reply:
Because the success field was never updated except calculateSuccess method. Inside calculateSuccess we performed (every time):
private void calculateSuccess() {
		success = !Iterables.any(entries, new IsErrorPredicate());
}
What's more, there was no getter for this field, so that means, if isSuccess method was called (and it was the only method to get success value), the field success was updated and calculated every time. @mjedraszczyk mentioned about that in his code review.